packages/ember-data/lib/system/store.js

```diff
@@ -1742,8 +1742,11 @@ Store = Service.extend({
 
   _modelForMixin: function(modelName) {
     var normalizedModelName = normalizeModelName(modelName);
-    var registry = this.container._registry ? this.container._registry : this.container;
-    var mixin = registry.resolve('mixin:' + normalizedModelName);
+    // container.registry = 2.1
+    // container._registry = 1.11 - 2.0
+    // container = < 1.11
+    var registry = this.container.registry || this.container._registry || this.container;
+    var mixin = this.container.lookupFactory('mixin:' + normalizedModelName);
```

Comment:
Need to double check that `container.lookupFactory` has existed since 1.8 before this can be merged.